### DISCUSSION TITLE
Eltako Switch not working via HA (*partly solved*)

### DISCUSSION THREAD

**Discussion by gsbox** [mak-gitdev/HA_enoceanmqtt#127] [General]
Hi.
I've got some FSR14-4 Devices in my Weberhaus, and i am trying to convert my Smarthome from FHEM to HA. I installed this addon and it seems to boot up just fine (logs attached).
I also wrote a script for the switch
`switch:
  - platform: template
    switches:
      az_licht:
        friendly_name: Arbeitszimmer Licht
        value_template: "{{ is_state('binary_sensor.e2m_az_licht_bo_pressed', 'on') }}"
        turn_on:
          - service: select.select_option
            data:
              option: BO
            target:
              entity_id: select.e2m_v_az_licht_action
        turn_off:
          - service: select.select_option
            data:
              option: BI
            target:
              entity_id: select.e2m_v_az_licht_action`

An i created the necessary devices (i hope) in enoceanmqtt.devices
`[az_licht]
address         = 0xFFD7BC11
rorg            = 0xF6
func            = 0x02
type            = 0x02.

[v_az_licht]
address 		= 0xFFFFFFFF
rorg    		= 0xF6
func    		= 0x02
type    		= 0x01
sender  		= 0xFF834381
virtual 		= 1`

When i turn the Switch, it stays there for approx. 2 seconds and then returns. The light stays off. Nothing to be seen, but the log seems to say, that the request was ok.
Is it a problem, that the sender-id of the virtual device was already in use through FHEM? This system is not running, while testing HA.

I'd really appreciate a hint, what i do wrong. From my humble opinion, there should be no problem in using the already working virtual-IDs from FHEM, right ?

Thanks for your help. If i can provide more detailed information, please let me know.
[enoceanmqtt.log](https://github.com/mak-gitdev/HA_enoceanmqtt/files/14250844/enoceanmqtt.log)

**Comment by mak-gitdev** [comment]
Hi @gsbox,

I had a quick look at the log and your template and everything seems to be ok.
The switch coming back to its initial position indicates that nothing is received from the FSR14 so the issue may arise from the sender ID used.
Was this sender ID used by FHEM to control the exact same light ?
If the answer is yes, then use another sender ID and everything should be OK.

Note that you can directly control your FSR14 instead of using virtual buttons using this other solution discussed [here](https://github.com/mak-gitdev/HA_enoceanmqtt/discussions/57#discussioncomment-8340082).
That solution sends Eltako GFVS telegrams to the devices and I think this is what FHEM actually does.

In my opinion, that's why you should not use the same sender ID as FHEM if you want to use virtual buttons, unless you reset the FSR14 and do the pairing again with the virtual button.

**Reply by gsbox** [reply]
Hey there.

Thanks a lot for your answer. I'll definetly try this out, once i'm back home. Yes. this is the same ID that i used in fhem. I thought, i might spare me the work to re-learn every switch and blind again, as i have already done in fhem y couple of years ago :)
And i thought that it doesn't matter, if this already learned id comes from fhem or HA. But i'll give it a try. And i will look into the menitioned discussion. Seems very promising. I hope, that i can take my long evolved FHEM-System to HA. 

I reply as soon as possible. Thanks

**Reply by mak-gitdev** [reply]
@gsbox,

Yes you're right regarding the ID. If FHEM was sending F6 telegrams like you are doing with virtual buttons, it would have worked smoothly with the same ID.
But FHEM, I think, sends GFVS telegrams which are different from F6 telegrams, and your ID has been paired with your FSR14 like this.
So when you send F6 telegram with the same FHEM ID, I think your FSR14 simply discards that telegram.

As the other solution aforementioned also sends GFVS telegrams, then I think you can use your FHEM ID and it should work without pairing again. Interesting use case this, please let me know :wink:

**Reply by sipodd** [reply]
@gsbox 
Translating to the Eltako PCT14 View:
FHEM, and the "beta" Version of this Addon need to be entered like this:
F4SR14: "senderadress" from device_file in Funktionsgruppe 2 - Funktion 51

The Virtual Switches, like the Eltako Switches you probably bought too, are "Funktion 2" 
I added a brief explanation [here](https://github.com/mak-gitdev/hidden-addon/discussions/2#discussioncomment-8434752)
So, if you update the addon, it should work as you expected.

**Reply by gsbox** [reply]
Hey @mak-gitdev
I want to thank you very much for your help. I installed the suggested Test-Addon and it works great to switch my FSR14 as well as my FSB14 shutters. Almost perfekt :+1: Now i have to find out, why the Status of the device is not (yet) displayed 

But i am glad it works now. 
Thanks a lot for your help. Its very much appreciated.

Thanks to @sipodd as well. Great work 

**Edit 1** : I got it working now (the lights). I just had the wrong ID .... 

Now i'm working on having the values for the shutter correctly and get a response from the fsb-Devices as well. I can control the blinds, but i do not have a status vom the device yet.

 
The only thing i'm a little bit concerned about is, that it is a test-repo... I hope it'll make it to a stable release soon. But as long as i can control my lights and blinds i am pretty happy to use this branch.

**Reply by gsbox** [reply]
Hi there.

I just wanted to tell you my final stage so far.

- FSR14 - Light switches work as charm and provide feedback as well. So perfektly working 
- FSB14 
   - The blinds are controllable (up / down / stop), but i don't see any information on where the blinds are (position-wise).  
   - The shutter-seconds seem to be ignored. It does not stop automatically
- FMS14 - is not yet accessible in HA. 

So my thoughts are, that i start to transfer my not eltko-related automations to HA. This is already taking some time. My lights and blinds are being controlled via HTTP-Api and hopefully one day will be completely supported in HA. I'll definitely keep an eye on the progress on this topic.

Thanks a lot for your support and your work, to get Eltako devices to HA

**Reply by mak-gitdev** [reply]
@gsbox,

With the current mapping for FSB14, when you use the UP and DOWN arrows, the shut time defined on the device is used to move the cover, not the one defined with the addon.
This is due to a limitation of the MQTT integration that prevents the use of templates in the payload_open (UP) and payload_close (DOWN) MQTT messages. So I can't set the current defined shut time.
But if you set the cover position instead of using UP and DOWN arrows to move the cover, then it should work well.
If you have an issue with the position, maybe your device is not configured to send a feedback telegram with the position info.
In that case, please share your log, with debug activated, when you move the cover up and down, so we can confirm this assumption.

Regarding FMS14, for the moment, you could use virtual buttons. I will later have a look for a better support.

**Reply by gsbox** [reply]
Ok. Thanks for the information. I'll try this and give you feedback. :+1: 

And regarding the virtual buttons, i do not exactly know, what you mean by this, but i have not yet searched for this topic. I am just at the beginning with HA, but i'll try to find this out. So it should already be possible, to control FMS14 ? That would  be great :+1:

**Reply by mak-gitdev** [reply]
@gsbox,

Sorry I meant virtual devices 😅, like you wanted to do initially.
